src/main/java/frc/robot/commands/auton/AutonBuilder.java

```diff
@@ -0,0 +1,280 @@
+    private final SwerveSubsystem swerveSubsystem;
+    private final LightBarSubsystem lightBarSubsystem;
+    private final FieldManagementSubsystem fmsSubsystem;
+    private final PIDController thetaController;
+    private final PIDController xPID;
+    private final PIDController yPID;
+
+    /** Constructs a {@link AutonBuilder} with auton-abstracted functions. */
+    public AutonBuilder(IntakePivotSubsystem intakePivotSubsystem,
+                             IntakeRollerSubsystem intakeRollersSubsystem,
+                             ShooterFlywheelSubsystem shooterFlywheelSubsystem,
+                             ShooterPivotSubsystem shooterPivotSubsystem,
+                             ElevatorSubsystem elevatorSubsystem,
+                             SwerveSubsystem swerveSubsystem,
+                             LightBarSubsystem lightBarSubsystem,
+                             FieldManagementSubsystem fmsSubsystem) {
+        this.intakePivotSubsystem = intakePivotSubsystem; 
+        this.intakeRollerSubsystem = intakeRollersSubsystem;
+        this.shooterFlywheelSubsystem = shooterFlywheelSubsystem;
+        this.shooterPivotSubsystem = shooterPivotSubsystem;
+        this.elevatorSubsystem = elevatorSubsystem;
+        this.swerveSubsystem = swerveSubsystem;
+        this.lightBarSubsystem = lightBarSubsystem;
+        this.fmsSubsystem = fmsSubsystem;
+
+        xPID = new PIDController(4, 0, 0);
+        yPID = new PIDController(4, 0, 0);
+        thetaController = new PIDController(3.5, 0, 0);
+    }
+
+    /**
+     * Follows trajectory.
+     *
+     * @param trajectory ChoreoTrajectory
+     * @return followPath command
+     */
+    public Command followPath(ChoreoTrajectory trajectory) {
+        return Choreo.choreoSwerveCommand(
+            trajectory,
+            swerveSubsystem::getRobotPosition,
+            xPID, 
+            yPID, 
+            thetaController, 
+            ((ChassisSpeeds speeds) -> {
+                swerveSubsystem.setChassisSpeeds(
+                    speeds.vxMetersPerSecond,
+                    speeds.vyMetersPerSecond, 
+                    speeds.omegaRadiansPerSecond
+                );
+            }),
+            fmsSubsystem::isRedAlliance,
+            swerveSubsystem
+        );
+    }
+
+    /**
+     * Follows trajectory to intake.
+     *
+     * @param intakeTrajectory ChoreoTrajectory
+     * @return goIntake Command
+     */
+    public Command goIntake(ChoreoTrajectory intakeTrajectory) {
+        return followPath(intakeTrajectory).alongWith(
+            new ElevatorToIntakeCommand(elevatorSubsystem).andThen(
+                new IntakePivotSetPositionCommand(intakePivotSubsystem, 1)
+            )
+        ).andThen(
+            new IntakeRollerIntakeCommand(intakeRollerSubsystem, lightBarSubsystem)
+                .raceWith(new DriveForwardCommand(swerveSubsystem).withTimeout(AutonConstants.INTAKE_SWERVE_TIME)),
```

Comment:
You can probably set the timeout on the whole thing instead? Because the way this `raceWith` is composed the entire thing exits when the timeout is up anyways.